### PR TITLE
Add post_process_semantic_segmentation method to DPTFeatureExtractor

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
Adds post_process_semantic_segmentation method to DPTFeatureExtractor.

I will open an issue and separate PRs to make sure that:
- Segmentation models (DETR, MaskFormer, SegFormer, etc.) have consistently named post-processing methods, arguments and outputs
- ImageSegmentationPipeline works with all available segmentation models

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/dpt/feature_extraction_dpt.py [hunk 1]
```diff
@@ -200,3 +202,44 @@ def __call__(
...
+    def post_process_semantic_segmentation(self, outputs, target_sizes: Union[TensorType, List[Tuple]] = None):
+        """
+        Args:
+        Converts the output of [`DPTForSemanticSegmentation`] into semantic segmentation maps. Only supports PyTorch.:
+            outputs ([`DPTForSemanticSegmentation`]):
+                Raw outputs of the model.
+            target_sizes (`torch.Tensor` of shape `(batch_size, 2)` or `List[Tuple]` of length `batch_size`,
+            *optional*):
+                Torch Tensor (or list) corresponding to the requested final size (h, w) of each prediction. If left to
+                None, predictions will not be resized.
```
Comment (sgugger) [inline src/transformers/models/dpt/feature_extraction_dpt.py:215]: Same problem as the other PRs.

File: src/transformers/models/dpt/feature_extraction_dpt.py [hunk 2]
```diff
@@ -200,3 +202,45 @@ def __call__(
...
+
+    def post_process_semantic_segmentation(self, outputs, target_sizes: Union[TensorType, List[Tuple]] = None):
+        """
+        Converts the output of [`DPTForSemanticSegmentation`] into semantic segmentation maps. Only supports PyTorch.
+
+        Args:
+            outputs ([`DPTForSemanticSegmentation`]):
+                Raw outputs of the model.
+            target_sizes (`torch.Tensor` of shape `(batch_size, 2)` or `List[Tuple]` of length `batch_size`,
+            *optional*):
```
Comment (sgugger) [inline src/transformers/models/dpt/feature_extraction_dpt.py:214]:
```suggestion
            target_sizes (`torch.Tensor` of shape `(batch_size, 2)` or `List[Tuple]` of length `batch_size`,0 *optional*):
```
This needs to be on the same line, as mentioned in the other PR.

File: src/transformers/models/dpt/feature_extraction_dpt.py [hunk 3]
```diff
@@ -200,3 +202,45 @@ def __call__(
...
+                Torch Tensor (or list) corresponding to the requested final size (h, w) of each prediction. If left to
+                None, predictions will not be resized.
+        Returns:
+            semantic_segmentation: `torch.Tensor` of shape `(batch_size, 2)` or `List[torch.Tensor]` of length
```
Comment (sgugger) [inline src/transformers/models/dpt/feature_extraction_dpt.py:218]: The type should be first
```
`torch.Tensor` or `List[torch.Tensor]`: tensor of shape `(batch_size, 2)` or list of tensors of length `batch_size`...
```

File: src/transformers/models/dpt/feature_extraction_dpt.py [hunk 4]
```diff
@@ -200,3 +202,45 @@ def __call__(
...
+            `batch_size`, where each item is a semantic segmentation map of of the corresponding target_sizes entry (w,
+            h) if `target_sizes` is specified). Each entry of each `torch.Tensor` correspond to a semantic class id.
```
Comment (sgugger) [inline src/transformers/models/dpt/feature_extraction_dpt.py:220]: Use width height and not w, h since we are in a docstring.

**Review by sgugger — APPROVED** [review]
Please make sure your docstrings are in the right format and don't contain any abbreviation :-)
LGTM otherwise.

**Review by NielsRogge — APPROVED** [review]
LGTM!

**Pull request merged**